### ISSUE TITLE
Recommended approach for provisioning during image creation (packer etc)

### ISSUE THREAD

**Issue by mihasya** [poise/citadel#16]
Hey there, we're not quite sure of the cleanest way to proceed with configuring Packer for running cookbooks containing `citadel` calls. We either have to have a dev bucket with dev secrets that we pass to packer (this will work, as we already have this for our dev environments, but it feels sloppy to bake these into an image), or exclude certain recipes/resources based on an attribute - lots of conditionals all over the place. Have you run into this issue, and how did you work around it?

One idea I had was adding a global "killswitch" attribute that could be set that would turn `citadel(..)` into a no-op. I've seen this pattern in some other cookbooks ([Pagerduty'd Sumologic CB](https://github.com/PagerDuty/chef-sumologic/blob/e50950ad43fb535e3680e1760464d7e7c8b97e01/libraries/sumo_source_provider.rb#L17), for example). If you're fine with this approach, I'll happily submit a PR. Thanks!

**Comment by coderanger** [comment]
I don't think it would make sense to put a killswtich in citadel itself, since it still has to return something no matter what. The right place for this switch is in your cookbooks that use citadel, where you can make it a true no-op.

**Comment by mihasya** [comment]
Noah, we actually ended up setting up a separate s3 bucket filled with
dev credentials and a special set of Iam creds that only can access that
bucket. Allows for more thorough testing anyway, so probably the better
approach. Sorry I had neglected to follow up on the issue.

On Thursday, May 28, 2015, Noah Kantrowitz <notifications@github.com> wrote:

> Closed #16 <https://github.com/poise/citadel/issues/16>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/poise/citadel/issues/16#event-316910596>.
>


-- 
Sent from Russia, with love. But also from my phone.
